opendatahub-io/odh-model-controller
# add 'bucket' parameter to storage-secret

## Description

Fixes #115 
Add `bucket` parameter to the storage-secret created by odh-model-controller. This is in addition to the already existing `default_bucket` parameter in the storage-secret. 
This is because kserve and modelmesh have slightly different formats for the secret. Kserve expects the bucket to be specified in  `bucket`, but ModelMesh expects the bucket to be specified in `default_bucket`. Long term solution should be to create an upstream issue to have the formats match upstream. For now, adding both keys as parameter having the same value should make the secret work for both kserve and modelmesh. 

## How Has This Been Tested?



- Local changes were built and pushed to `quay.io/vedantm/odh-model-controller:storage_secret_fix` 
- Created a branch `storage_secret_fix_test` which modifies `params.env` to deploy `quay.io/vedantm/odh-model-controller:storage_secret_fix`  
- Tested changes using the custom image and branch. 

### Testing instructions : 
- Install Kserve dependencies 
- Install ODH operator 2.3+ 
- Create the following DSC 
```
cat <<EOF |oc create -f - 
spec:
  components:
    dashboard:
      devFlags:
        manifests:
          - contextDir: manifests
            sourcePath: ''
            uri: >-
              https://github.com/opendatahub-io/odh-dashboard/tarball/f/model-serving
      managementState: Managed
    kserve:
      devFlags:
        manifests:
          - contextDir: config
            sourcePath: overlays/odh
            uri: 'https://github.com/opendatahub-io/kserve/tarball/release-v0.11.1'
          - contextDir: config
            sourcePath: ''
            uri: >-
              https://github.com/VedantMahabaleshwarkar/odh-model-controller/tarball/storage_secret_fix_test
      managementState: Managed
EOF
``` 
- Log in to ODH Dashboard
- Create a Data Science Project `test` 
- Create Data Connection called `rhodss3` for the `rhods-public` s3 bucket (it's a public bucket so your own aws access+secret key should work, region is us-east-1)
- Create ServingRuntime 
```
cat <<EOF |oc create -f - 
apiVersion: serving.kserve.io/v1alpha1
kind: ServingRuntime
metadata:
  name: caikit-runtime
spec:
  containers:
  - env:
    - name: RUNTIME_LOCAL_MODELS_DIR
      value: /mnt/models
    # Last tested image stable-27c0d61
    image: quay.io/opendatahub/caikit-tgis-serving:stable   
    name: kserve-container
    ports:
    # Note, KServe only allows a single port, this is the gRPC port. Subject to change in the future
    - containerPort: 8085
      name: h2c
      protocol: TCP
    resources:
      requests:
        cpu: 4
        memory: 8Gi
  multiModel: false
  supportedModelFormats:
  # Note: this currently *only* supports caikit format models
  - autoSelect: true
    name: caikit
EOF
```
- Create InferenceService 
```
cat <<EOF |oc create -f - 
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  annotations:
    serving.knative.openshift.io/enablePassthrough: "true"
    sidecar.istio.io/inject: "true"
    sidecar.istio.io/rewriteAppHTTPProbers: "true"
  name: caikit-example-isvc
spec:
  predictor:
    model:
      modelFormat:
        name: caikit
      runtime: caikit-runtime
      storage:
        key: aws-connection-rhodss3
        path: demo-models/flan-t5-small-caikit/flan-t5-small-caikit/
EOF
```
- Check isvc successfully deploys 

## Merge criteria:
